spec/requests/api/alerts_spec.rb

```diff
@@ -44,4 +44,96 @@
       "id"   => alert_status.id
     )
   end
+
+  context "alert_actions subcollection" do
+    let(:alert) { FactoryGirl.create(:miq_alert_status) }
+    let(:actions_subcollection_url) { "#{alerts_url(alert.id)}/alert_actions" }
+
+    it "forbids access to alerts actions subcolletion without an appropriate role" do
+      FactoryGirl.create(
+        :miq_alert_status_action,
+        :miq_alert_status => alert,
+        :user             => FactoryGirl.create(:user)
+      )
+      api_basic_authorize
+      run_get(actions_subcollection_url)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it "reads an alert action as a sub collection under an alert" do
+      api_basic_authorize subcollection_action_identifier(:alerts, :alert_actions, :read, :get)
+      alert_action = FactoryGirl.create(
+        :miq_alert_status_action,
+        :miq_alert_status => alert,
+        :user             => FactoryGirl.create(:user)
+      )
+      run_get(actions_subcollection_url)
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to eq(
+        "name"      => "alert_actions",
+        "count"     => 1,
+        "subcount"  => 1,
+        "resources" => [
+          {
+            "href" => "http://www.example.com/api/alerts/#{alert.id}/alert_actions/#{alert_action.id}"
+          }
+        ]
+      )
+    end
+
+    it "forbids creation of an alert action under alerts without an appropriate role" do
+      api_basic_authorize
+      run_post(
+        actions_subcollection_url,
+        "action_type" => "comment",
+        "comment"     => "comment text",
+      )
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it "creats an alert action under an alert" do
+      user = FactoryGirl.create(:user)
+      attributes = {
+        "action_type" => "comment",
+        "user_id"     => user.id,
+        "comment"     => "comment text",
+      }
+      api_basic_authorize subcollection_action_identifier(:alerts, :alert_actions, :create, :post)
+      run_post(actions_subcollection_url, attributes)
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body["results"].first).to have_attributes(attributes)
```

Comment:
This is fine, but in the case that it breaks you may get a NoMethodError on `nil`, instead of a diff. So it may be better to write these and others like it something like:


expected = {
  "results" => [
    a_hash_including(attributes)
  ]
}
expected(response.parsed_body). to include(expected)
```